install_files/securedrop-app-code/DEBIAN/postinst

```diff
@@ -16,8 +17,55 @@ set -x
 # for details, see http://www.debian.org/doc/debian-policy/ or
 # the debian-policy package
 
+database_migration() {
+    database_dir='/var/lib/securedrop'
+    database_file="$database_dir/db.sqlite"
+
+    # Ensure the database exists.
+    if [ ! -e "$database_file" ]; then
+        old_umask=$(umask)
+        umask 0027 # to get os 0640 permissions from the sqlite command
+
+        # Future-proof way to create an empty database.
+        sqlite3 "$database_file" '.databases'
+        umask "$old_umask"
+        chown www-data:www-data "$database_file"
+    fi
+
+    cd '/var/www/securedrop/'
+
+    # `alembic current` outputs nothing if the DB isn't under verion control.
+    if [ -z "$(alembic current)" ]; then
+        sqlite_tables=$(sqlite3 "$database_file" .tables)
+
+        if ! echo "$sqlite_tables" | grep -q journalists; then
+            # This is a fresh install, run the full migrations.
+            alembic upgrade head
+        else
+            # This is an old install, so mark that the scheme is up to date
+            # as of the version of migration 15ac9509fc68_init.py when we first
+            # started version controlling the database
+            alembic stamp 15ac9509fc68
```

Comment:
What do you mean "template this out"?

Reply:
Thinking something like this so its not hard-coded. This value will theoretically change frequently right? If my understanding is incorrect then I retract my comment :)

https://github.com/freedomofpress/securedrop/blob/4bd7d8f218317083427dfc6ee994c6a673795f05/install_files/securedrop-grsec/DEBIAN/control.j2#L3

Reply:
Ah. No, we want to hard coded because the migration needs to always start at the `init` revision. This line only happens if the database is empty *and* not under version control yet.